### PR TITLE
Add QMK support for Red Scarf I numberpads

### PR DESCRIPTION
**Pull request by Defying**
## Description
Adds support for AVR-based Red Scarf I numberpads. Availability is limited, was a small group buy on geekhack years ago. Works perfectly on QMK!

## Types of Changes
Added a new keyboard using the `util/new_keyboard.sh` script, and tested until fully working to my best ability.

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* n/a

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/redscarf_i/readme.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Red Scarf I
+
+![redscarf_i](https://i.imgur.com/Q6p81aX.jpg)
+
+Port of QMK for Red Scarf I number pads. Default keymaps designed for ortho layouts.
+
+Keyboard Maintainer: [Ben Weakley](https://github.com/defying)  
+Hardware Supported: Red Scarf I PCB's
```
Comment (fauxpark) [inline keyboards/redscarf_i/readme.md]:
```suggestion
Hardware Supported: Red Scarf I PCB's  
```

File: keyboards/redscarf_i/redscarf_i.c [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  setPinOutput(F6);
+  setPinOutput(F5);
+  // and then turn them off
+  writePinHigh(F7);
+  writePinHigh(F6);
+  writePinHigh(F5);
+}
+
+void led_set_kb(uint8_t usb_led) {
+  if (usb_led & (1<<USB_LED_NUM_LOCK)) {
```
Comment (fauxpark) [inline keyboards/redscarf_i/redscarf_i.c]:
```suggestion
  if (IS_LED_ON(usb_led, USB_LED_NUM_LOCK)) {
```

File: keyboards/redscarf_i/redscarf_i.h [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT_4x6_ortho( \
```
Comment (fauxpark) [inline keyboards/redscarf_i/redscarf_i.h]: This layout should be called `LAYOUT_ortho_6x4` to support the Community layouts feature.
It also would be a good idea to add the numpad layouts here (`numpad_6x4` and `numpad_5x4`).

```suggestion
#define LAYOUT_ortho_6x4( \
```

File: keyboards/redscarf_i/redscarf_i.h [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+{ \
+    { k00, k01, k02, k03 }, \
+    { k10, k11, k12, k13 }, \
+    { k20, k21, k22, k23 }, \
+    { k30, k31, k32, k33 }, \
+    { k40, k41, k42, k43 }, \
+    { k50, k51, k52, k53 }  \
+}
+
+#define LAYOUT_4x5_ortho( \
```
Comment (fauxpark) [inline keyboards/redscarf_i/redscarf_i.h]:
```suggestion
#define LAYOUT_ortho_5x4( \
```

File: keyboards/redscarf_i/rules.mk [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+BOOTLOADER = atmel-dfu
+
+# build options
+BOOTMAGIC_ENABLE = no   # Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE  = no   # Mouse keys(+4700)
+EXTRAKEY_ENABLE  = yes  # Audio control and System control(+450)
+CONSOLE_ENABLE   = no   # Console for debug(+400)
+COMMAND_ENABLE   = no   # Commands for debug and configuration
+BACKLIGHT_ENABLE = yes  # Enable keyboard backlight functionality
+UNICODE_ENABLE   = no   # Unicode
```
Comment (fauxpark) [inline keyboards/redscarf_i/rules.mk]:
```suggestion
UNICODE_ENABLE   = no   # Unicode

LAYOUTS = ortho_5x4 ortho_6x4 numpad_5x4 numpad_6x4
```

**Comment by fauxpark** [comment]
Also, this board is missing a `default` keymap.

**Comment by drashna** [comment]
Any updates on this?

File: keyboards/redscarf_i/readme.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+Hardware Availability: no longer available
+
+Make example for this keyboard (after setting up your build environment):
+
+    make redscarf_i:4x6_ortho:dfu
```
Comment (fauxpark) [inline keyboards/redscarf_i/readme.md]:
```suggestion
    make redscarf_i:4x6_ortho
```

File: keyboards/redscarf_i/rules.mk [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# MCU options
+MCU = atmega32u4
+BOOTLOADER = atmel-dfu
```
Comment (fauxpark) [inline keyboards/redscarf_i/rules.mk]:
```suggestion

# Bootloader selection
#   Teensy       halfkay
#   Pro Micro    caterina
#   Atmel DFU    atmel-dfu
#   LUFA DFU     lufa-dfu
#   QMK DFU      qmk-dfu
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
BOOTLOADER = atmel-dfu
```

File: keyboards/redscarf_i/rules.mk [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = atmel-dfu
+
+# build options
+BOOTMAGIC_ENABLE = no   # Virtual DIP switch configuration(+1000)
```
Comment (fauxpark) [inline keyboards/redscarf_i/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE = no   # Virtual DIP switch configuration
```

File: keyboards/redscarf_i/rules.mk [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+MOUSEKEY_ENABLE  = no   # Mouse keys(+4700)
```
Comment (fauxpark) [inline keyboards/redscarf_i/rules.mk]:
```suggestion
MOUSEKEY_ENABLE  = no   # Mouse keys
```

File: keyboards/redscarf_i/rules.mk [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+EXTRAKEY_ENABLE  = yes  # Audio control and System control(+450)
+CONSOLE_ENABLE   = no   # Console for debug(+400)
```
Comment (fauxpark) [inline keyboards/redscarf_i/rules.mk]:
```suggestion
CONSOLE_ENABLE   = no   # Console for debug
```

**Comment by Defying** [comment]
Working on that, sorry for all the delays! I'm new to this and sometimes forgetful. 😀

File: keyboards/redscarf_i/rules.mk [hunk 11]
```diff
@@ -0,0 +1,23 @@
+# MCU selection
```
Comment (fauxpark) [inline keyboards/redscarf_i/rules.mk]:
```suggestion
# MCU name
```

File: keyboards/redscarf_i/keymaps/ortho_custom_4x5/keymap.c [hunk 12]
```diff
@@ -0,0 +1,35 @@
+#include QMK_KEYBOARD_H
+
+enum custom_keycodes {
+  KC_P00 = SAFE_RANGE,
+  MR_0001,
+  MR_0002
```
Comment (drashna) [inline keyboards/redscarf_i/keymaps/ortho_custom_4x5/keymap.c]: These appear to not be used?

File: keyboards/redscarf_i/redscarf_i.h [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+    k30, k31, k32, k33, \
+    k40, k41, k42, k43  \
+) \
+{ \
+    { k00, k01, k02, k03 }, \
+    { k10, k11, k12, k13 }, \
+    { k20, k21, k22, k23 }, \
+    { k30, k31, k32, k33 }, \
+    { k40, k41, k42, k43 }  \
+}
```
Comment (noroadsleft) [inline keyboards/redscarf_i/redscarf_i.h]: You need layout macros for `LAYOUT_numpad_6x4` and `LAYOUT_numpad_5x4` in this file also.

**Pull request closed without merging**